### PR TITLE
Add Low pT Occupancies

### PR DESCRIPTION
**Pull request by GNiendorf**
This PR adds the occupancies for the 0.6 GeV pT threshold. It changes the current if statements for the occupancies into a 2d matrix of category and eta number for cleaner code. There are 2 matrices for each object, the current 0.8 GeV occupancies matrix and an additional matrix for the 0.6 GeV occupancies. This PR also cleans up some code copy-paste related to the occupancy kernels for selecting the relevant eta and category bins. Lastly, I also added the script that was used to select the 0.6 GeV occupancies.

### CODE REVIEW

File: RecoTracker/LSTCore/src/alpaka/MiniDoublet.h [hunk 1]
```diff
@@ -1007,76 +1008,71 @@ namespace SDL {
       nTotalMDs = 0;
       alpaka::syncBlockThreads(acc);
 
-      // Initialize variables outside of the for loop.
-      int occupancy, category_number, eta_number;
+      // Occupancy matrix for 0.8 GeV pT Cut
+      constexpr int p08_occupancy_matrix[4][4] = {
+          {49, 42, 37, 41},      // category 0
+          {100, 100, 100, 100},  // category 1
```
Comment (GNiendorf) [inline RecoTracker/LSTCore/src/alpaka/MiniDoublet.h:1014]: It seems like the last two occupancies in category 1 go unused, unless I made a mistake somewhere.

Comment (GNiendorf) [reply RecoTracker/LSTCore/src/alpaka/MiniDoublet.h]: I fixed this in my commit below. I haven't checked how much memory it saves though compared to the "master" branch.

**Comment by GNiendorf** [comment]
@ariostas Is there any way you could add an optional parameter to the CI to test with the low pT cut? The only thing that would have to change is adding a -p 0.6 for the standalone code, although I'm not sure how you would update the CMSSW run.

**Comment by ariostas** [comment]
> Is there any way you could add an optional parameter to the CI to test with the low pT cut?

Yeah, I'll figure out how one is supposed to adjust parameters in CMSSW and I'll add the option to the CI.

**Comment by ariostas** [comment]
/run all lowpt

It should work now unless I messed something up. You'll just have to add "lowpt" somewhere in the comment, and if you want to test both setups then you'll just have to make two comments. For now, I didn't make the pt cut configurable (so only 0.8 and 0.6), but if there's a need I can do that.

**Comment by ariostas** [comment]
/run all lowpt

(I did mess something up. Trying again.)

**Comment by GNiendorf** [comment]
@ariostas Thank you for updating the CI! @slava77 I think this PR is ready for review when you get a chance.

File: RecoTracker/LSTCore/src/alpaka/MiniDoublet.h [hunk 2]
```diff
@@ -994,11 +995,42 @@ namespace SDL {
     }
   };
 
+  // Helper function to determine eta bin for occupancies
+  ALPAKA_FN_ACC ALPAKA_FN_INLINE int getEtaBin(const float module_eta) {
+    if (module_eta < 0.75f)
+      return 0;
+    if (module_eta < 1.5f)
```
Comment (slava77) [inline RecoTracker/LSTCore/src/alpaka/MiniDoublet.h:1002]:
```suggestion
    else if (module_eta < 1.5f)
```
since we tend to get these in the PR reviews

**Review by slava77 — COMMENTED** [review]
(common comment)

File: RecoTracker/LSTCore/src/alpaka/MiniDoublet.h (hunk 2 above)
Comment (GNiendorf) [reply RecoTracker/LSTCore/src/alpaka/MiniDoublet.h]: Fixed in latest commit.

**Comment by slava77** [comment]
> Here is a timing comparison:

pLS time went up from 505 to 551.
Is this real? (I can't quite match with the changes in the code)

@GNiendorf 
please run the timing locally to check.
Thank you.

**Comment by GNiendorf** [comment]
@slava77 I don't see any timing difference locally

Current CPU timing (4555, 7f3fe16b8e3):
master_timing

This PR CPU timing:
my_pr_timing

Current GPU timing (4555, 7f3fe16b8e3):
Screenshot 2024-07-05 at 4 29 27 PM

This PR GPU timing:
Screenshot 2024-07-05 at 4 32 18 PM

**Review by slava77 — APPROVED** [review]

**Pull request merged**